internal/cloudapi/server.go

```diff
@@ -521,3 +522,87 @@ func (server *Server) GetVersion(w http.ResponseWriter, r *http.Request) {
 		panic("Failed to write response")
 	}
 }
+
+// ComposeMetadata handles a /compose/{id}/metadata GET request
+func (server *Server) ComposeMetadata(w http.ResponseWriter, r *http.Request, id string) {
+	jobId, err := uuid.Parse(id)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Invalid format for parameter id: %s", err), http.StatusBadRequest)
+		return
+	}
+
+	var result worker.OSBuildJobResult
+	status, _, err := server.workers.JobStatus(jobId, &result)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Job %s not found: %s", id, err), http.StatusNotFound)
+		return
+	}
+
+	var job worker.OSBuildJob
+	if _, _, _, err = server.workers.Job(jobId, &job); err != nil {
+		http.Error(w, fmt.Sprintf("Job %s not found: %s", id, err), http.StatusNotFound)
+		return
+	}
+
+	var rpms []rpmmd.RPM
+	if status.Finished.IsZero() {
+		// job still running: empty response
+		if err := json.NewEncoder(w).Encode(new(ComposeMetadata)); err != nil {
+			panic("Failed to write response: " + err.Error())
+		}
+		return
+	}
+	rpms = rpmmd.OSBuildStagesToRPMs(result.OSBuildOutput.Build.Stages)
```

Comment:
This is wrong.  It's not the PRMs for the build root that we want but the RPMs for the payload.

Reply:
I should also add a test for this.  A check to verify that a non-standard package specified in the request is included in the metadata.

Reply:
Fixed.